mne_lsl/stream/_base.py

```diff
@@ -846,15 +948,20 @@ def _pick(self, picks: NDArray[+ScalarIntType]) -> None:
             )
 
         with self._interrupt_acquisition():
-            with use_log_level(logger.level):
-                self._info = pick_info(self._info, picks)
+            self._info = pick_info(self._info, picks, verbose=logger.level)
             self._picks_inlet = self._picks_inlet[picks_inlet]
             self._buffer = self._buffer[:, picks]
-
             # prune added channels which are not part of the inlet
             for ch in self._added_channels[::-1]:
                 if ch not in self.ch_names:
                     self._added_channels.remove(ch)
+            # remove dropped channels from filters
+            for filt in self._filters:
+                # TODO: ensure correct selection of channels.
+                filt["picks"] = np.intersect1d(filt["picks"], picks, assume_unique=True)
+                # TODO: don't reset, select initial conditions.
+                filt["zi"] = None
```

Comment:
Not outdated, the link you shared estimates the initial `zi` response for the `picks.size` channels to which this filter is applied. This `zi` will be of shape `(..., n_picks)`.
This comment lies in the `_pick` method which is called when channels are dropped from the Stream. What I did here for now is that if channels are dropped, then all filters have their initial conditions 'reset' (and thus will be re-estimated during the next online iteration). Else, if a filter had one of the channel it applies on removed, the `n_picks` shape of the `zi` becomes erroneous. 
Instead of resetting all filters (or resetting filters which have their `picks` modified), I would like to pick also along the last axes of the initial conditions `zi`, ensuring a smoother transition when dropping channels.

Reply:
By the way, the `picking` code on filters in this PR is likely wrong anyway, I haven't looked at this part yet ;)